contrib/opencensus-ext-requests/opencensus/ext/requests/trace.py

```diff
@@ -127,14 +159,37 @@ def wrap_session_request(wrapped, instance, args, kwargs):
     except Exception:  # pragma: NO COVER
         pass
 
-    # Add the requests url to attributes
-    _tracer.add_attribute_to_current_span(HTTP_URL, url)
+    # Add the requests host to attributes
+    _tracer.add_attribute_to_current_span(
+        HTTP_HOST, dest_url)
 
-    result = wrapped(*args, **kwargs)
+    # Add the requests method to attributes
+    _tracer.add_attribute_to_current_span(
+        HTTP_METHOD, method.upper())
 
-    # Add the status code to attributes
+    # Add the requests path to attributes
     _tracer.add_attribute_to_current_span(
-        HTTP_STATUS_CODE, str(result.status_code))
+        HTTP_PATH, path)
+
+    # Add the requests url to attributes
+    _tracer.add_attribute_to_current_span(HTTP_URL, url)
 
-    _tracer.end_span()
-    return result
+    try:
+        result = wrapped(*args, **kwargs)
+    except requests.Timeout:
+        _span.set_status(exceptions_status.TIMEOUT)
```

Comment:
@reyang  Isn't this `except` clause and the ones following going to eat the raised exception and returns `None` in case of any `requests` error? Unless I'm missing something this change is going to break many things.

Reply:
Yes, but wont `requests` return `None` in case it raises an `Exception`? If so, the returned value is the same

Reply:
Raising an exception should not "return" at all.

```python
try:
  requests.get("http://not-a-host")
  unreacheable() #actually reached with this PR
except Exception:
  reached() # actually unreachable with this PR
```

Reply:
What can be changed, and is a good idea actually, is to threat the `Exception` and raise it again, and let the user do whatever it wants. That's what you meant?

Reply:
This should be fixed by adding a `raise` statement in every `except` block.

Reply:
The main goal here is to keep this hook transparent and keep 1:1 interface as `requests` (Exceptions are part of that interface).

Reply:
> What can be changed, and is a good idea actually, is to threat the Exception and raise it again, and let the user do whatever it wants. That's what you meant?

So you agreed with this, right? Just to be clear. If so, I do agree with you and I will open a PR to change this.

Reply:
Yes :+1: !

Reply:
@isra17 You're right, we've missed this. It is a bug.